### PR TITLE
2020 workshop agenda rough cut

### PR DESCRIPTION
**Pull request by hjoliver**
Quick brain dump to get the ball rolling

(Suggest additions or changes by review here, or merge now then and post change PRs)

### CODE REVIEW

File: docs/feb2020-workshop-agenda.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Primarily: review Cylc 8 progress to date and map out our roadmap to
+  completion
+- Secondarily: learn as much as we can about parts of the system mostly
+  worked on by others, to gain understanding and enable future work
+
+## Prelimary Agenda Topics
+
+(TBD: collate with others' suggestions and make a timetable)
+
+- architecture
```
Comment (dpmatthews) [inline docs/feb2020-workshop-agenda.md]: Not sure if this is the right section but I think we should include some discussion around how we support cli access via the hub (including for task communication purposes)

File: docs/feb2020-workshop-agenda.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - other views: dot, graph, etc.
+    - when to start on these?
+  - common data store and subscriptions etc.
+
+- spawn-on-demand enhancement
+  - POC implemented already, solves many problems
+  - consequences (e.g. need n=1,2,3 workflow window)
+  - for Cylc 8?
+
+- platform changes
```
Comment (dpmatthews) [inline docs/feb2020-workshop-agenda.md]: Discuss potential user & global config changes (going back to the suggestions made during the aborted unified config work). We'll try to prepare some notes / a proposal in advance.
(Not really related to platform changes so probably needs a separate topic)

File: docs/feb2020-workshop-agenda.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+- rose suite-run migration and related changes including new "cylc run"
+  semantics (name/run1,2,3... etc.)
+
+- authorization
+  - can we settle on an initial set of privilege levels?
+  - implementation ideas
+
```
Comment (dpmatthews) [inline docs/feb2020-workshop-agenda.md]: Add a topic on release & deployment strategy? e.g.
* versioning strategy (now we have many repos)
* future of cylc wrapper / multi-version support?
* minimal client install?
* use of `conda pack`?
* reducing the size of conda environments?
* optional dependencies?
* installing without conda?

**Comment by kinow** [comment]
As it hasn't been merged yet, maybe we can add a space for a discussion around containers?

File: docs/feb2020-workshop-agenda.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+- practical debugging session
+  - how to debug the UI (with browser and Vue dev tools)
+  - how to debug WFS and the UIS
+
+- coding: can we usefully work together on something, for several afternoons
```
Comment (oliver-sanders) [inline docs/feb2020-workshop-agenda.md]: Perhaps scope for a whiteboard/inkscape design session:

* Gantt view.
* Multi-workflows / dashboards.
* Selecting sharing (drag n drop?).
* Other?

File: docs/feb2020-workshop-agenda.md (hunk 3 above)
Comment (oliver-sanders) [reply docs/feb2020-workshop-agenda.md]: (adding to Dave's comment):

* reducing size of UI `dist/` package.

File: docs/feb2020-workshop-agenda.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    - when to start on these?
+  - common data store and subscriptions etc.
+
+- spawn-on-demand enhancement
+  - POC implemented already, solves many problems
+  - consequences (e.g. need n=1,2,3 workflow window)
+  - for Cylc 8?
+
+- platform changes
+
```
Comment (oliver-sanders) [inline docs/feb2020-workshop-agenda.md]: - Might be worth having a brief discussion about contingency planning for a delayed Rose2 release - New Python2 Rose release with `rose suite-run` etc stripped out for use with Cylc8.

File: docs/feb2020-workshop-agenda.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
+## Cylc Workshop
+### 10-14 Feb 2020, NIWA, Wellington, NZ
+
+#### Attendees:
+
+- Hilary Oliver, NIWA
```
Comment (wxtim) [inline docs/feb2020-workshop-agenda.md]: Don't forget to change this if we have any extra people like American Tim

File: docs/feb2020-workshop-agenda.md (hunk 4 above)
Comment (wxtim) [reply docs/feb2020-workshop-agenda.md]: I think that another useful topic would be to tackle the `cylc go` (working title) CLI in the way you did the GUI. I write the output of `cylc go --help` and we can shred it together.

File: docs/feb2020-workshop-agenda.md (hunk 2 above)
Comment (wxtim) [reply docs/feb2020-workshop-agenda.md]: Happy to look back and summarize proposals made by people in various PRs and issues.

**Comment by wxtim** [comment]
@hjoliver suggest that you respond as you see fit and merge, leaving it to others to make PRs about their own topics. Else this could get messy.

**Comment by hjoliver** [comment]
Thanks all, will briefly attempt to add bits as suggested and merge before the meeting in 30 min...

File: docs/feb2020-workshop-agenda.md (hunk 1 above)
Comment (hjoliver) [reply docs/feb2020-workshop-agenda.md]: I think you mean UI Server, not Hub, right?

Comment (dpmatthews) [reply docs/feb2020-workshop-agenda.md]: I assume you have to go via Hub to get to the UI server?

**Pull request merged**